tests/tasks/setup_squid.yml

```
@@ -0,0 +1,88 @@
+# SPDX-License-Identifier: MIT
+---
+- name: Install and configure squid
+  when:
+    - lsr_rhc_test_data.proxy_noauth_hostname == "127.0.0.1"
+  block:
+    - name: Install packages for squid
+      package:
+        name:
+          - squid
+          - httpd-tools
+        state: present
+
+    - name: Check the status of the backup of configuration
+      stat:
+        path: /etc/squid/squid.conf.BACKUP
+      register: squid_conf_backup_stat
+
+    - name: Backup the configuration
+      copy:
+        src: /etc/squid/squid.conf
+        dest: /etc/squid/squid.conf.BACKUP
+        remote_src: true
+        mode: 0644
+      when:
+        - not squid_conf_backup_stat.stat.exists
+
+    - name: Copy the pristine configuration back
+      copy:
+        src: /etc/squid/squid.conf.BACKUP
+        dest: /etc/squid/squid.conf
+        remote_src: true
+        mode: 0644
+      when:
+        - squid_conf_backup_stat.stat.exists
+
+    - name: Open the Candlepin port
+      lineinfile:
+        path: /etc/squid/squid.conf
+        regexp: "^acl SSL_ports port {{ lsr_rhc_test_data.candlepin_port }}"
+        insertbefore: "^acl Safe_ports "
+        line: >-
+          acl SSL_ports port {{ lsr_rhc_test_data.candlepin_port }} # Candlepin
+
+    - name: Setup no authentication
+      when:
+        - authentication | d(false) == false
```

```suggestion
        - not (authentication | d(false))
```
ansible/jinja style is not to do comparisons to `true` or `false`